src/front-end/typescript/lib/pages/proposal/code-with-us/edit/index.tsx

```diff
@@ -185,10 +185,21 @@ function makeComponent<K extends Tab.TabId>(): component_.page.Component<
               const proposal = proposalResponse.value;
               const opportunity = opportunityResponse.value;
               const affiliations = affiliationsResponse.value;
+              const [sidebarState, sidebarCmds] = Tab.makeSidebarState(
+                tabId,
+                proposal.id,
+                opportunity.id
+              );
               const tabComponent = Tab.idToDefinition(tabId).component;
               return [
-                state.set("proposal", proposal),
+                state
+                  .set("proposal", proposal)
+                  .set("sidebar", immutable(sidebarState)),
                 [
+                  ...component_.cmd.mapMany(
+                    sidebarCmds,
+                    (msg) => adt("sidebar", msg) as Msg
+                  ),
```

Comment:
Wasn't sure exactly where I needed to set the immutable `sidebarState`, but this is how the opportunities page does it so I assume it's fine?

Reply:
The sidebar state doesn't need to be re-initialized here. We typically want to re-initialize sidebar state if it requires the result of the async `Cmd`s that make network requests. For the CWU proposal edit page, the `makeSidebarState` function (https://github.com/bcgov/digital_marketplace/blob/d9e48a69290175dfe73edf9f24665d9404496f26/src/front-end/typescript/lib/pages/proposal/code-with-us/edit/tab/index.ts#L95) doesn't need this data.

FYI looks like the bug was caused by `sidebarState` not being made immutable in `init` (see L95: https://github.com/bcgov/digital_marketplace/blob/d9e48a69290175dfe73edf9f24665d9404496f26/src/front-end/typescript/lib/pages/proposal/code-with-us/edit/index.tsx#L95).

Reply:
Yeah, I thought about doing that in L95 for that file; good to know that should patch it up and thanks for the explanation r.e. re-initializing the sidebar.